sklearn/metrics/tests/test_common.py

```diff
@@ -1550,12 +1549,13 @@ def check_sample_weight_invariance(name, metric, y1, y2):
     ),
 )
 def test_regression_sample_weight_invariance(name):
-    n_samples = 50
-    random_state = check_random_state(0)
+    n_samples = 51
```

Comment:
This cannot be an even number due to #17370, inspired by @ogrisel last paragraph in https://github.com/scikit-learn/scikit-learn/issues/17370#issuecomment-2413282001, even though I agree it's a temporary fix.

Reply:
Since #29907 was merged, it should be possible to update `median_absolute_error` to handle even number of data points (with or without weights) as explained below.